pimd/pim_register.c

```diff
@@ -263,9 +266,70 @@ void pim_null_register_send(struct pim_upstream *up)
 			return;
 		}
 	}
-	pim_register_send((uint8_t *)&ip_hdr, sizeof(struct ip),
-			src, rpg, 1, up);
+	pim_register_send((uint8_t *)&ip_hdr, sizeof(struct ip), src, rpg, 1,
+			  up);
 }
+#else
+void pim_null_register_send(struct pim_upstream *up)
+{
+	struct ip6_hdr ip6_hdr;
+	struct pim_msg_header pim_msg_header;
+	struct pim_interface *pim_ifp;
+	struct pim_rpf *rpg;
+	pim_addr src;
+	unsigned char buffer[100];
+	struct ipv6_ph ph;
+
+	pim_ifp = up->rpf.source_nexthop.interface->info;
+	if (!pim_ifp) {
+		if (PIM_DEBUG_PIM_TRACE)
+			zlog_debug(
+				"%s: Cannot send null-register for %s no valid iif",
+				__func__, up->sg_str);
+		return;
+	}
+
+	rpg = RP(pim_ifp->pim, up->sg.grp);
+	if (!rpg) {
+		if (PIM_DEBUG_PIM_TRACE)
+			zlog_debug(
+				"%s: Cannot send null-register for %s no RPF to the RP",
+				__func__, up->sg_str);
+		return;
+	}
+
+	memset(&ip6_hdr, 0, sizeof(struct ip6_hdr));
+	ip6_hdr.ip6_nxt = PIM_IP_PROTO_PIM;
+	ip6_hdr.ip6_plen = PIM_MSG_HEADER_LEN;
+	ip6_hdr.ip6_hlim = 1;
+	ip6_hdr.ip6_vfc = 6;
+	ip6_hdr.ip6_src = up->sg.src;
+	ip6_hdr.ip6_dst = up->sg.grp;
+
+	memset(buffer, 0, 10000);
```

Comment:
10000? Can we use something more readable?

Reply:
And, it's initialized to 100, which means overflow?

Reply:
Corrected. Missed to update the size in memset. Corrected now